hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java

```diff
@@ -100,4 +105,45 @@ static void destroyPipeline(DatanodeDetails dn, PipelineID pipelineID,
           true, p.getId());
     }
   }
+
+  static int encodeNodeIdsOfFactorThreePipeline(List<DatanodeDetails> nodes) {
+    if (nodes.size() != HddsProtos.ReplicationFactor.THREE.getNumber()) {
+      return 0;
+    }
+    List<UUID> nodeIds = nodes.stream()
+        .map(DatanodeDetails::getUuid).distinct()
+        .collect(Collectors.toList());
+    nodeIds.sort(new ComparableComparator());
```

Comment:
Can we use orderless hash here to avoid sorting? E.g., xor the hash of all the nodeId hash?

Reply:
I will use xor then